templates/FFV_form_factors.cpp.in

```diff
@@ -429,14 +479,91 @@ double OneLoopFunctionN(double r, double s)
       } else if (is_zero(1.0 - r)){
          return 2.0 / 3.0;
       } else {
-         return (1.0 - 4.0 * r + 3 * r * r - 2 * r * r * std::log(r)) / Power3(1.0 - r);
+         return (1.0 - 4.0 * r + 3 * r * r - 2 * r * r * std::log(r)) / 2.0 / Power3(1.0 - r);
       }
+   } else if (1./s < ratiolimit) {
+      // Expand s about 0 to 1st order
+      return (1.0 - r * r * s + r * (s - 1.0) + r * (s + 1.0) * std::log(r)) / Power2(r - 1.0);
    } else {
       return s * (r * Power2(r - s) * std::log(r) + (r - 1.0) * ((r - s) * (s - 1.0) - (r - 1.0)
                               * r * std::log(r/s))) / Power2(1.0 - r) / Power2(r - s) / (s - 1.0);
    }
 }
 
+
+double Kallenlambda(double m1, double m2, double m3)
+{
+    return Power2(m1) + Power2(m2) + Power2(m3) - 2.*m1*m2 - 2.*m2*m3 - 2.*m3*m1;
+}
+
+double Kallenlambdasqrt(double m1, double m2, double m3)
+{
+    return std::sqrt(Power2(m1) + Power2(m2) + Power2(m3) - 2.*m1*m2 - 2.*m2*m3 - 2.*m3*m1);
+}
+
+double DiscontinuityB(double m12, double m2, double m3)
+{
+    return Kallenlambdasqrt(m12,Power2(m2),Power2(m3)) 
+	    * std::log((Power2(m2)+Power2(m3)-m12+Kallenlambdasqrt(m12,Power2(m2),Power2(m3)))/(2.*m2*m3)) / m12;
+}
+
+double signeddilog(double x, double a)
+{
+    if (a > 0.)
+    {
+        return -1./2. * Power2(std::log(1./(1.-x))) - dilog(x/(x-1.));
+    } else if (a < 0.) {
+        return dilog(x);
+    }
+}
+
+double ScalarC0(double M22, double M32, double m1, double m2, double m3)
```

Comment:
Would you mind adding a reference for this function, maybe to a paper or so, in form of a comment? It seems to me that this is the classic `C0` Passarino-Veltman function. If so, we may move this function into `src/numerics.h`.

Reply:
This is the C0 Passarino-Veltman function with one of the mass entries assumed to be zero (the mass of the outgoing photon).

Reply:
Ah, I see. If it is a very special case, we can leave it here for the moment.